tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/BlockGridEditor/Datatype/BlockGridEditorDataTypeBlocks.spec.ts

```diff
@@ -0,0 +1,738 @@
+    await umbracoApi.documentTypes.ensureNameNotExists(elementNameTwo);
+    await umbracoApi.documentTypes.ensureNameNotExists(elementNameThree);
+
+    const elementOne = await umbracoApi.documentTypes.createDefaultElementType(elementName, elementAlias);
+    const elementTwo = await umbracoApi.documentTypes.createDefaultElementType(elementNameTwo, elementTwoAlias);
+    const elementThree = await umbracoApi.documentTypes.createDefaultElementType(elementNameThree, elementThreeAlias);
+
+    const blockGridType = new BlockGridDataTypeBuilder()
+      .withName(blockGridName)
+      .addBlockGroups()
+        .withName(groupOne)
+      .done()
+      .addBlockGroups()
+        .withName(groupTwo)
+      .done()
+      .addBlock()
+        .withContentElementTypeKey(elementOne['key'])
+      .done()
+      .addBlock()
+        .withContentElementTypeKey(elementTwo['key'])
+        .withGroupName(groupOne)
+      .done()
+      .build();
+    await umbracoApi.dataTypes.save(blockGridType);
+
+    await umbracoUi.navigateToDataType(blockGridName);
+
+    // Adds another element to GroupTwo
+    // We need to have a nth because all the add block groups are the same in the html
+    await page.locator('[key="blockEditor_addBlockType"]').nth(2).click();
+    await umbracoUi.clickDataElementByElementName("tree-item-" + elementNameThree);
+    await umbracoUi.clickElement(umbracoUi.getButtonByLabelKey(ConstantHelper.buttons.submitChanges));
+    await umbracoUi.clickElement(umbracoUi.getButtonByLabelKey(ConstantHelper.buttons.save));
+
+    // Assert
+    await umbracoUi.isSuccessNotificationVisible();
+    // Checks if the elements is are added to their correct groups
+    await expect(page.locator('.umb-block-card-group').nth(0).locator('[data-content-element-type-key="' + elementOne['key'] + '"]')).toBeVisible();
+    await expect(page.locator('.umb-block-card-group').nth(1).locator('[data-content-element-type-key="' + elementTwo['key'] + '"]')).toBeVisible();
+    await expect(page.locator('.umb-block-card-group').nth(2).locator('[data-content-element-type-key="' + elementThree['key'] + '"]')).toBeVisible();
+    // Checks if the datatype was created
+    await expect(await umbracoApi.dataTypes.exists(blockGridName)).toBe(true);
+    await umbracoUi.doesDataTypeExist(blockGridName);
+
+    // Clean
+    await umbracoApi.documentTypes.ensureNameNotExists(elementName);
+    await umbracoApi.documentTypes.ensureNameNotExists(elementNameTwo);
+    await umbracoApi.documentTypes.ensureNameNotExists(elementNameThree);
+  });
+
+  test('can create a block grid datatype with multiple groups and multiple element in each group', async ({page, umbracoApi, umbracoUi}) => {
+    const elementNameTwo = 'SecondElement';
+    const elementTwoAlias = AliasHelper.toAlias(elementNameTwo);
+    const elementNameThree = 'ThirdElement';
+    const elementThreeAlias = AliasHelper.toAlias(elementNameThree);
```

Comment:
These should not be function variables, but rather class variables, so we don't have to define them in every test.